### ISSUE TITLE
Add better subtitle support to the generators

### ISSUE THREAD

**Issue by IncontinentCell** [mortii/anki-morphs#297]
### Is your feature request related to a problem?

Currently only some formats of files are supported by the generators. Namely ".txt" ".srt" ".vtt" ".md". Based on my experience ".ass" files are often used for subtitles and currently they cannot be loaded. I've also noticed slight issue with the current way ".srt" files are parsed.

[//]: # (Please provide a clear and concise description of problem if it exists.)

### Describe the solution you would like

[//]: # (Please provide a clear and concise description of the solution.)

Adding support of ".ass" files to the generators. Properly parsing subtitle files instead of just iterating over their file's text lines.

### Additional context

[//]: # (Add any additional context about the problem here, or delete this section if it's empty)

I've taken a quick look at AnkiMorphs' code and it seems file parsing is done by this function for all file formats:
```python
def create_file_morph_occurrences(
    preprocess_options: PreprocessOptions,
    file_path: Path,
    file_io: TextIO,
    morphemizer: Morphemizer,
    nlp: Any,
) -> dict[str, MorphOccurrence]:
    # nlp: spacy.Language

    all_lines: list[str] = []
    morph_occurrences: dict[str, MorphOccurrence]

    try:
        for line in file_io:
            # lower-case to avoid proper noun false-positives
            filtered_lines = filter_line(preprocess_options, line=line.lower())
            all_lines.append(filtered_lines)
    except UnicodeDecodeError as exc:
        raise UnicodeException(path=file_path) from exc

    if nlp is not None:
        morph_occurrences = get_morph_occurrences_by_spacy(
            preprocess_options, nlp, all_lines
        )
    else:
        morph_occurrences = get_morph_occurrences_by_morphemizer(
            preprocess_options, morphemizer, all_lines
        )

    return morph_occurrences
```

All we do is take lines from the file, filter them and extract the morphs. 
This approach works for plain text files, but has issues with ".srt"
Consider the following fragment from an example srt subtitle:
```
3
00:00:12,637 --> 00:00:17,517
何かを得ようと欲すれば必ず同等の
対価を支払うものなり
```
The entire text is a single sentence, but the create_file_morph_occurrences function will separate those into two lines:
"何かを得ようと欲すれば必ず同等の" and "対価を支払うものなり"
This can cause the generated morphs to be incorrect as the sentences fed into the Morphemizer are often split while they shouldn't be. 
I think for subtitle files it'd be better to properly parse them instead of doing this.

There are many python libraries to handle parsing subtitles. For example this one could be used:
[pysub-parser](https://pypi.org/project/pysub-parser/) - Supports ass, ssa, srt, sub

**Comment by mortii** [comment]
Sure, I can take another stab at adding `.ass` support. 

> There are many python libraries to handle parsing subtitles.

The worst part about creating anki add-ons is that you can't import/use non-standard libraries very easily, hence the nightmare process of getting spacy to work.

Regardless, like you highlighted, the current process is fairly naive so it's definitely due for an upgrade.

**Comment by mortii** [comment]
Released in [v3.3.0](https://github.com/mortii/anki-morphs/releases/tag/v3.3.0), thanks!
